### PR TITLE
Retail autoaccept keys

### PR DESCRIPTION
**Pull request by Loquacity**
# Description

Adds new section to Deploy Terminals - Other section for auto-accepting keys with grain, plus some drive by edits in the Deploy Terminals chapter.

# Target branches

Which product version does this PR apply to?

- [ ] Manager-4.0
- [ ] Manager-4.1
- [x] Manager-4.2
- [x] Uyuni

# Links

Fixes https://github.com/SUSE/spacewalk/issues/12393

### CODE REVIEW

File: modules/retail/pages/retail-deploy-terminals-other.adoc [hunk 1]
```diff
@@ -61,7 +61,7 @@ For more information about the Saltboot formula, see xref:salt:formula-saltboot.
 
 
 
-== Bootstrap Terminals over a Wireless Network
+== Deploy Terminals over a Wireless Network
 
 For terminals that cannot be connected directly to the physical network, you can bootstrap them over a wireless network.
```
Comment (keichwa) [inline modules/retail/pages/retail-deploy-terminals-other.adoc:66]:
```suggestion
For terminals that cannot be connected directly to the physical network, you can deploy them over a wireless network.
```
to be consistent with the title ;)  drive-by edit

File: modules/retail/pages/retail-deploy-terminals-other.adoc [hunk 2]
```diff
@@ -212,3 +212,94 @@ Then you can specify a second Salt state with the wireless details for use durin
...
+====
+
+
+To do this, you first need to configure Saltboot, then configure the {productname} Server.
+You can configure Saltboot to automatically sign grains once, or set it to always auto-sign.
+Alternatively, you can configure Saltboot during PXE boot using kernel parameters.
+
+
+.Procedure: Configuring Saltboot to Auto-Sign Once
+. On the branch server, create a configuration file called [path]``./etc/salt/minion.d/autosign-grains-onetime.conf``.
```
Comment (keichwa) [inline modules/retail/pages/retail-deploy-terminals-other.adoc:237]:
```suggestion
. On the branch server, create a configuration file called [path]``/etc/salt/minion.d/autosign-grains-onetime.conf``.
```

File: modules/retail/pages/retail-deploy-terminals-other.adoc [hunk 3]
```diff
@@ -212,3 +212,94 @@ Then you can specify a second Salt state with the wireless details for use durin
...
+    autosign_key: <AUTOSIGN_KEY>
+
+# send the grain as part of auth request
+autosign_grains:
+    - autosign_key
+----
+. At the command prompt, add the new configuration file to the existing ``initrd``:
++
+----
+echo ./etc/salt/minion.d/autosign-grains-onetime.conf | /
```
Comment (keichwa) [inline modules/retail/pages/retail-deploy-terminals-other.adoc:253]:
```suggestion
echo /etc/salt/minion.d/autosign-grains-onetime.conf | /
```

File: modules/retail/pages/retail-deploy-terminals-other.adoc (hunk 2 above)
Comment (keichwa) [reply modules/retail/pages/retail-deploy-terminals-other.adoc:237]: I'm not sure about this suggestion, but the current path looks strange.

File: modules/retail/pages/retail-deploy-terminals-other.adoc [hunk 4]
```diff
@@ -212,3 +212,94 @@ Then you can specify a second Salt state with the wireless details for use durin
...
+
+
+.Procedure: Configuring Saltboot to Auto-Sign During PXE Boot
+. Configure the PXE formula to specify these kernel parameters during booting:
++
+----
+SALT_AUTOSIGN_GRAINS=autosign_key:<AUTOSIGN_KEY>
+----
+. PXE boot the Salt client.
+  The formula creates the [path]``./etc/salt/minion.d/autosign-grains-onetime.conf`` configuration file and passes it to ``initrd``.
```
Comment (keichwa) [inline modules/retail/pages/retail-deploy-terminals-other.adoc:283]:
```suggestion
  The formula creates the [path]``/etc/salt/minion.d/autosign-grains-onetime.conf`` configuration file and passes it to ``initrd``.
```

**Review by keichwa — APPROVED** [review]

File: modules/retail/pages/retail-deploy-terminals-other.adoc (hunk 2 above)
Comment (nadvornik) [reply modules/retail/pages/retail-deploy-terminals-other.adoc:237]: This is method 2 above.
The idea is to start in any working directory, create the directory structure and the file and append it to initrd.
It should not change anything in /etc on branch server.

The same procedure is already documented in wireless section above

So the path is correct.

File: modules/retail/pages/retail-deploy-terminals-other.adoc (hunk 3 above)
Comment (nadvornik) [reply modules/retail/pages/retail-deploy-terminals-other.adoc:253]: no, this is correct

File: modules/retail/pages/retail-deploy-terminals-other.adoc [hunk 5]
```diff
@@ -212,3 +212,94 @@ Then you can specify a second Salt state with the wireless details for use durin
...
+----
+echo ./etc/salt/minion.d/autosign-grains-onetime.conf | /
+cpio -H newc -o | gzip >> /srv/saltboot/boot/initrd.gz
+----
+
+
+
+.Procedure: Configuring Saltboot to Always Auto-Sign
+. On the Salt client, as root, create a configuration file called [path]``etc/salt/minion.d/autosign-grains.conf``.
+. Edit the new configuration file with these details.
```
Comment (nadvornik) [inline modules/retail/pages/retail-deploy-terminals-other.adoc:261]: I think this is meant to be the method 1 above, so this is not "Always Auto-Sign" and it should describe modifying image sources (on build host or in git) before building.

Because the way how the images are built, the file is first included in system image and then copied to intrd, so the grains are always visible in system image - so this method works only for autosign-grains.conf, not for autosign-grains-onetime.conf

File: modules/retail/pages/retail-deploy-terminals-other.adoc (hunk 4 above)
Comment (nadvornik) [reply modules/retail/pages/retail-deploy-terminals-other.adoc:283]: This is not correct. The formula passes the option to saltboot initrd.
Saltboot initrd creates /etc/salt/minion.d/autosign-grains-onetime.conf in the intrd during boot.

File: modules/retail/pages/retail-deploy-terminals-other.adoc [hunk 6]
```diff
@@ -212,3 +212,94 @@ Then you can specify a second Salt state with the wireless details for use durin
...
+
+
+.Procedure: Configuring the Server to Auto-Accept
+. On the {productname} Server, open the master configuration file in the [path]``/etc/salt/master.d/`` directory, and add or edit this line:
++
+----
+autosign_grains_dir: /etc/salt/autosign_grains
+----
+. Create a file at [path]``/etc/salt/autosign_grains``, that contains the autosign key you specified with Saltboot:
++
```
Comment (nadvornik) [inline modules/retail/pages/retail-deploy-terminals-other.adoc:298]: ``/etc/salt/autosign_grains`` is a directory.
The file is named after the grain
``/etc/salt/autosign_grains/<grain_name>``

So, to be consistent with the examples above, it should be
``/etc/salt/autosign_grains/autosign_key``

**Review by nadvornik — CHANGES_REQUESTED** [review]

**Review by jcayouette — APPROVED** [review]

**Comment by keichwa** [comment]
Vladimir Nadvornik <notifications@github.com> writes:

> So the path is correct.

Thanks for clarification, Vladimir!  Lana, please ignore my suggestions.

-- 
Karl Eichwalder

File: modules/retail/pages/retail-deploy-terminals-other.adoc [hunk 7]
```diff
@@ -212,3 +212,94 @@ Then you can specify a second Salt state with the wireless details for use durin
...
+
+[IMPORTANT]
+====
+Automatically accepting keys is less secure than manually checking and accepting keys.
+Only use this method on trusted networks.
+====
+
+
+To do this, you first need to configure Saltboot, then configure the {productname} Server.
+You can configure Saltboot to automatically sign grains once, or set it to always auto-sign.
```
[the remark this answers is not in the document]
Comment (Loquacity) [reply modules/retail/pages/retail-deploy-terminals-other.adoc:232]: updated

File: modules/retail/pages/retail-deploy-terminals-other.adoc (hunk 5 above)
Comment (Loquacity) [reply modules/retail/pages/retail-deploy-terminals-other.adoc:261]: updated step 1

**Review by nadvornik — CHANGES_REQUESTED** [review]
I don't see any changes regarding my previous feedback.

**Review by keichwa — APPROVED** [review]

**Comment by Loquacity** [comment]
> I don't see any changes regarding my previous feedback.

Sorry! Looks like I forgot to sync :woman_facepalming:

**Comment by nadvornik** [comment]
The titles should be changed, it should be clear that there are 3 methods how to add the autosign grains to saltboot:
- append to existing initrd
- modify image sources
- use kernel commandline via pxe formula

If the grain is kept or not, this is less important detail. It should be mentioned, but not in the title.

**Comment by Loquacity** [comment]
> The titles should be changed, it should be clear that there are 3 methods how to add the autosign grains to saltboot:
> 
> * append to existing initrd
> * modify image sources
> * use kernel commandline via pxe formula
> 
> If the grain is kept or not, this is less important detail. It should be mentioned, but not in the title.

These titles talk about the action, not the result.

File: modules/retail/pages/retail-deploy-terminals-other.adoc [hunk 8]
```diff
@@ -212,3 +212,102 @@ Then you can specify a second Salt state with the wireless details for use durin
...
+The auto-signed grain is then stored on the client as a regular Salt grain after booting.
+
+Alternatively, you can configure Saltboot during PXE boot using kernel parameters.
+
+
+
+.Procedure: Configuring Saltboot to Send Auto-Signed Grain Once
+. On the branch server, create a configuration file called [path]``/etc/salt/minion.d/autosign-grains-onetime.conf``.
+. Edit the new configuration file with these details.
+  You can use any value you like as the autosign key:
```
Comment (keichwa) [inline modules/retail/pages/retail-deploy-terminals-other.adoc:247]: wondering whether we should write auto-sign (with a dash)?  Here, and more often.

**Review by keichwa — APPROVED** [review]
